### DISCUSSION TITLE
How are "true wind" characteristics calculated?

### DISCUSSION THREAD

**Discussion by Lincoln-Xcel** [NREL/flasc#209] [Q&A] [answered]
Hello,
I am new to Flasc and trying to go through the examples with my own data. I saw within the example SCADA data from the folder 01_raw_data_processing, there are three columns labeled wd_truth, ws_truth, and ti_truth. I assume these are referring to true wind characteristics and I am curious if these values are a given in some SCADA data or calculated somehow using the data from the individual turbines?
Thank you for your help!

**Comment by misi9170** [comment, marked as answer]
Hi @Lincoln-Xcel ,

Usually, as you are hinting, the true wind direction, wind speed, and TI are not available (possibly if there are any met mast channels available, that could be used if they are trustworthy). Instead, we often end up taking data from one or more "reference turbines" as the "true" wind speed, direction, etc. 

Determining good reference turbines can be a little tricky. We usually select either a single free stream turbine, or all of the free stream turbines, for a given wind direction to provide the reference wind direction and wind speed. You'll notice that that is a bit circular, particular with regards to wind direction---you need the wind direction to determine which turbines are free stream, but you don't know the wind direction until you know which turbines are free stream. However, the wind direction measurement of turbines is not as sensitive to being waked as the wind speed or turbulence intensity, so we have found that it is safe to used waked turbines for a best guess at wind direction, at least initially*.

FLASC includes a function for [determining the freestream turbines](https://nrel.github.io/flasc/_autosummary/flasc.utilities.floris_tools.get_upstream_turbs_floris.html#flasc.utilities.floris_tools.get_upstream_turbs_floris). 

As I said though, this is nuanced and we choose different references in different situations, based on unique characteristics in each wind farm/field campaign. I'd certainly be happy to discuss more---feel free to email me at michael.sinner@nrel.gov if you'd like!

Cheers,
Misha

*That the wind directions of all turbines are reasonably trustworthy assumes that all turbines have their northing calibration correct (i.e., the nacelle yaw position is north when it reads north). Often, this is not the case---we have seen a range of northing biases from smallish offsets to what appear to be completely random offsets. FLASC also has a workflow to help identify northing biases. See [here for the API documentation](https://nrel.github.io/flasc/_autosummary/flasc.data_processing.energy_ratio_wd_bias_estimation.html#module-flasc.data_processing.energy_ratio_wd_bias_estimation) for that module of the code, and [here for an example of the northing correction process](https://github.com/NREL/flasc/blob/main/examples_artificial_data/01_raw_data_processing/01_northing_calibration.ipynb) using an artificial dataset.

**Comment by Lincoln-Xcel** [comment]
Misha,
Thank you very much for the insight. This was very thorough and helpful; I will reach out if I encounter further questions.
Thanks again,
-Lincoln
